nsmtesting/nsmtesting.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package nsmtesting provides k8s specific testing asserts
+package nsmtesting
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/edwarnicke/exechelper"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"k8s.io/client-go/kubernetes"
+)
+
+// NSMTesting is k8s specific asset tool
+type NSMTesting struct {
+	t      *testing.T
+	client *kubernetes.Clientset
+}
+
+// New creates new NSMTesting
+func New(t *testing.T) *NSMTesting {
+	return &NSMTesting{
+		t: t,
+	}
+}
+
+// K8s returns k8s client
+func (t *NSMTesting) K8s() *kubernetes.Clientset {
+	return t.client
+}
+
+// Exec executes command
+func (t *NSMTesting) Exec(cmd string) {
```

Comment:
I still don’t understand why *NSMTestung exists... it simply obscures the action to save a single line of code inline in the tessuite itself... why not just roll it up into the Test*

Reply:
I think we need to use something like `NSMTesting` because we need to simplify using of exechelper. For example, let's take a look at the source code of the test based on your suggestion.

```go 
	writer := logrus.StandardLogger().Writer()
	s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/alpine.yaml", exechelper.WithStderr(writer), exechelper.WithStdout(writer)))
	defer s.Require().NoError(exechelper.Run("kubectl delete -f ./deployments/alpine.yaml", exechelper.WithStderr(writer), exechelper.WithStdout(writer)))
	s.Require().NoError(exechelper.Run("kubectl wait --for=condition=ready pod -l app=alpine", exechelper.WithStderr(writer), exechelper.WithStdout(writer)))
```

I think it looks more complex than
```
	s.Exec("kubectl apply -f ./deployments/alpine.yaml")
	defer s.Exec("kubectl delete -f ./deployments/alpine.yaml")
	s.Exec("kubectl wait --for=condition=ready pod -l app=alpine")
```

Reply:
@edwarnicke 

Can we add to exechelpder something like _instance_. For example:
```
executor := exechelpder.New(option1, option2)
...
executor.Run("echo ...") // == exechelpder.Run("echo ...", option1, option2)
```
Then we can remove NSMTesting  without loss of readability :)

What do you think?

Reply:
What about something like this:

```go
opts := []*exechelper.Option {
    exechelper.WithStderr(logrus.StandardLogger().Writer()),
    exechelper.WithStdout(logrus.StandardLogger().Writer()),
}
s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/alpine.yaml"),opts...)
defer s.Require().NoError(exechelper.Run("kubectl delete -f ./deployments/alpine.yaml", opts...)
s.Require().NoError(exechelper.Run("kubectl wait --for=condition=ready pod -l app=alpine", opts...)
```

From this you can basically see:

1.  What are these 'opts' (stdout and stderr to logging)
2.  What each line is doing (run this cmd with those opts

Reply:
(and if you'd like to move opts out of band... you could always define:

```go
func LogOutput() *exechelper.Option {
    return []*exechelper.Option {
    exechelper.WithStderr(logrus.StandardLogger().Writer()),
    exechelper.WithStdout(logrus.StandardLogger().Writer()),
}
```

and have 

```go
s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/alpine.yaml"),LogOutput()...)
defer s.Require().NoError(exechelper.Run("kubectl delete -f ./deployments/alpine.yaml", LogOutput()...)
s.Require().NoError(exechelper.Run("kubectl wait --for=condition=ready pod -l app=alpine", LogOutput()...)
```

Reply:
Thanks! Fixed.